scripts/update_data_with_components/update_data_with_components.gml

```diff
@@ -0,0 +1,5 @@
+function update_data_with_components(data)
+{
+	data.components = get_components(data.str);
+	data.plaintext = string_join_ext("", array_map(data.components, get_component_contents));
```

Comment:
I think this could be
```
function(component) {
   return component.contents;
});
```
And then, we can remove the global function `get_component_contents`, as it seems to be only used once.